modin/backends/base/query_compiler.py

```diff
@@ -2360,7 +2360,7 @@ def groupby_agg(
         """
         if isinstance(by, type(self)) and len(by.columns) == 1:
             by = by.columns[0] if drop else by.to_pandas().squeeze()
-        elif isinstance(by, type(self)):
+        elif drop and isinstance(by, type(self)):
```

Comment:
converting QC 'by' to a list of column labels only if this 'by' comes from the `self` (if `drop is True`)

Reply:
this would make an awesome one-line comment here explaining the "why this is done"

Reply:
done